crates/typst/src/foundations/calc.rs

```diff
@@ -908,6 +910,82 @@ pub fn quo(
     Ok(floor(dividend.apply2(divisor.v, Div::div, Div::div)))
 }
 
+/// Calculates the p-norm of a sequence of values.
+///
+/// ```example
+/// #calc.norm(1, 2, -3, 0.5)
+/// #calc.norm(p: 3, 1in, 2cm)
+/// #calc.norm(3em, 4em)
+/// ```
+#[func(title = "P-Norm")]
+pub fn norm(
+    /// The p value to calculate the p-norm of.
+    #[named]
+    #[default(Num::Int(2))]
+    p: Num,
+    /// The sequence of values from which to calculate the p-norm.
+    /// Returns `0.0` if empty.
+    #[variadic]
+    values: Vec<Spanned<Value>>,
+) -> SourceResult<Value> {
+    let mut sum = 0.0;
+    if let Some(Spanned { v, span }) = values.first() {
+        match v {
+            Value::Int(_) | Value::Float(_) => {
```

Comment:
This function could benefit from an enum similar to `AngleLike`.

Reply:
Any idea what it should be called?

Reply:
Maybe just something norm-specific like `Normable`? Maybe the maths folks have a better term for that.

Reply:
Normable is in fact used in math.

(Though in the sense that a topological vector space is normable if the topology can be induced by a norm.)